doc/modules/preprocessing_targets.rst

```diff
@@ -27,17 +30,49 @@ matrix from a list of multi-class labels::
     array([[1, 0, 0, 0],
            [0, 0, 0, 1]])
 
-For multiple labels per instance, use :class:`MultiLabelBinarizer`::
+.. warning::
 
-    >>> lb = preprocessing.MultiLabelBinarizer()
-    >>> lb.fit_transform([(1, 2), (3,)])
-    array([[1, 1, 0],
-           [0, 0, 1]])
-    >>> lb.classes_
-    array([1, 2, 3])
+    Label indicator matrices are intended for multilabel classification
+    (rather than multiclass classification). However, LabelBinarizer
+    repurposes this format for multiclass data to enable multiclass
+    classification in estimators that only support binary data.
+    Depending on your situation, you could consider using the
+    :term:`multiclass` target representation instead for estimators which
+    support other types of data.
```

Comment:
Hmm I'm not completely clear on what you are saying here. Are you just suggesting to keep data in multiclass format when the estimator supports multiclass classification? If so maybe be more explicit, on first reading as it sounds a little like we are suggesting to transform to multiclass instead but we are saying they don't need to transform (as label binarizer is multiclass -> multilabel, so data would be multiclass to start with)

Reply:
> Are you just suggesting to keep data in multiclass format when the estimator supports multiclass classification? If so maybe be more explicit, on first reading as it sounds a little like we are suggesting to transform to multiclass instead but we are saying they don't need to transform (as label binarizer is multiclass -> multilabel, so data would be multiclass to start with)

Thank you, this is exactly it. You've worded things in a much clearer way here.